### ISSUE TITLE
Time format in patient information

### ISSUE THREAD

**Issue by JackMatthams** [nhsuk/nhsuk-service-manual-backlog#167]

**Comment by sarawilcox** [comment]
We recently tested various time formats on assistive technology. Our findings suggests that the content style guide is correct about:
-  avoiding 1730 hours, because we saw examples of this being read out as 1 thousand, seven hundred 30 hours
- using "to" rather than a dash or hyphen in time ranges, because we saw examples of a dash either being ignored or being read out as "minus"
- using am and pm rather than the 24 hour clock

There was some evidence that a colon worked better than a full stop which can be read out as "point". 

I'm proposing to put a recommendation to the next Content Style Council that we use the format 8am to 6:30pm. 

But do please let us know if you find out anything else. 

Here's what we say in the content style guide: https://beta.nhs.uk/service-manual/content/numbers-measurements-dates-time#time

**Comment by Fenwick17** [comment]
As part of the profiles editor, we were looking into how we can allow users to input time.

I did some testing on using `<input type="time" />` but unfortunately that tested very poorly with VoiceOver on iOS, so unfortunately that can't be used. 
We have decided to go with using standard `<input type="text" />` and use content to guide users into the time format, which would be 24 hour clock stopping at 23:59.
We will also provide extensive form field validation to ensure users meet this criteria.

Users on the public facing side of the service will see times in the Gov standard of am/pm midday/midnight.

We will be conducting some user research on this in future.

**Comment by sarawilcox** [comment]
@rhiannonsmith Did you do any further research into this?

**Comment by rhiannonsmith** [comment]
@sarawilcox No, sorry we haven't had a chance to do any further research into this

**Comment by sarawilcox** [comment]
Note, for system date-times and time-stamps, e.g. for data exchange or data sorting, the NHS service standard refers to the GOV.UK guidance here: https://www.gov.uk/government/publications/open-standards-for-government/date-times-and-time-stamps-standard

**Comment by Fenwick17** [comment]
I have recently done some testing with VoiceOver around how times are read out using the NHS format of 5.30pm compared to the govuk format of 5:30pm
With VoiceOver the NHS method is read out as '5 decimal 30pm' compared to the gov style of '5 colon 30pm'. 
Both formats read out fairly similar, but generally I would go with the colon approach as I feel it is more common and because of that could cause less confusion.

**Comment by sarawilcox** [comment]
A suggestion via Slack that we make it clearer in the style guide that whole hours are written without a full stop (or colon), e.g. 10pm.

**Comment by OliviaThornton** [comment]
Is there any update on this issue? 
We use the format non-24hr clock with colon in the app (6:30pm, 6am), as listed under time in [GOV.UK A to Z](https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style) and would really like to see this in the service manual as a consistent time format.

**Comment by sarawilcox** [comment]
I think we can bring to this to an early Style Council meeting for a decision.

**Comment by sarawilcox** [comment]
Sorry we haven't got this onto a Style Council agenda yet @OliviaThornton. I'm noting some related tickets we should look at first:
- [Time input on GOV.UK Design System backlog](https://github.com/alphagov/govuk-design-system-backlog/issues/173)
- [Booking an appointment with a small time slot in NHS.UK backlog](https://github.com/nhsuk/nhsuk-service-manual-backlog/issues/270)

**Comment by sarawilcox** [comment]
In [this comment about some screenshots from NBS user research show and tell, March 2021](https://github.com/nhsuk/nhsuk-service-manual-backlog/issues/366#issuecomment-843279669) on the GitHub issue for texts and emails, the team found this:
![Screenshot 2021-05-19 at 09 08 17](https://user-images.githubusercontent.com/45792959/118778227-c926fe80-b881-11eb-9b76-d7088d8e1f1c.png)

**Comment by sarawilcox** [comment]
We've had a report via the service manual Slack of an issue with using a full stop rather than a colon for time slots, 
1.50pm instead of 1:50pm for example.

> "We've had a scenario come up where someone's device tries to get the time period to create a calendar entry but the period throws it off. The effect here is that the patient cannot easily add this item to their calendar (they can manually off course).
> 
> Since the time-formatting guideline was written, has there been any follow up to figure out if that is still the best option, given it seems it's almost standard elsewhere to use a colon instead?"

![image (48)](https://user-images.githubusercontent.com/45792959/224960654-c19ae05a-fb12-449a-af92-8cd23e5642aa.png)

**Comment by ZeldaRhiando-nhs** [comment]
I feel like we should follow the time format tested and validated by GDS, so:

hh:mm:ss

We could be using aria / <time> instead of <span> which would mean it was read out without ‘colon’, but of course only if it’s correctly formatted, with a colon, not a period:
https://stackoverflow.com/questions/48758106/how-do-i-indicate-to-a-screen-reader-that-a-time-is-hours-and-minutes-not-minut

**Comment by anupshah** [comment]
In the above (thanks @sarawilcox for raising my example) it is an SMS, so we don't have ability to hint using aria, or using html time element with date time attribute to make up for it. Using colon would be my vote too.

I didn't realise NHS guidance on time had deviated from GDS. Was there a specific scenario that was addressing?

**Comment by sarawilcox** [comment]
### Notes from June 2023 Style Council meeting

There's a difference between our approach to time (with a full stop) and GOV.UK's (with a colon) and that's confusing some content designers. 

NHS 5.30pm 

GOV 5:30pm 

We’ve had several requests over the years to come into line with GOV. Some parts of the NHS website (e.g. GP profiles) already use a colon.  

More recently we’ve also seen that our current style can raise interoperability issues. For example, in text messages, using a full stop stops users adding a calendar appointment. (Thanks for raising this @anupshah.) 

Colons are pretty standard now, for example, in Android and iPhone locked screens.  

In addition to the requests that we come into line with GOV and use a colon, we've also had some feedback that our guidance isn't clear on how to write whole hours. We propose to address both issues in the following change to the style guide. 

| We use: | We do not use: |
| 5pm | 5.00 pm or 1700hrs |
| 5:30pm | 5.30pm or 17.00hrs |

Action: approved. Highlighted text to be added to style guide, subject to clinical approval. 

There was some discussion about whether we need to clarify our guidance around midnight.  

We currently say to use “midday” or “midnight”. This seems to work in a general way, e.g. when a service is always closed at midnight. But Brian reported that some users get confused about which day “midnight” relates to. We also saw this in some COVID services. Needs more research. 

Action: Sara to liaise with Brian and report at next meeting. 

In fact, we already have a [GitHub issue for midnight](https://github.com/nhsuk/nhsuk-service-manual-community-backlog/issues/268) and GOV.UK has guidance on it.
